requirements/AUR-CBI-INT-0070/standalone-project/src/application/main.c

```diff
@@ -0,0 +1,266 @@
+    payload_flow_rate = 4.0;
+    payload_target_flow_rate = 0.0;
+
+    payload_step_count = 0;
+    payload_amp_100 = 0.0;
+    payload_amp_50 = 0.0;
+    payload_amp_25 = 0.0;
+}
+
+void application_step()
+{
+    if (electric_power_supply_step_count == 0) {
+        electric_power_supply_amp_40 = 0.8 * (rand() / (RAND_MAX - 2.0) - 1.0);
+    }
+    if (electric_power_supply_step_count % 20 == 0) {
+        electric_power_supply_amp_20 = 0.6 * (rand() / (RAND_MAX - 2.0) - 1.0);
+    }
+    if (electric_power_supply_step_count % 10 == 0) {
+        electric_power_supply_amp_10 = 0.4 * (rand() / (RAND_MAX - 2.0) - 1.0);
+    }
+    if (electric_power_supply_step_count % 5 == 0) {
+        electric_power_supply_amp_5 = 0.2 * (rand() / (RAND_MAX - 2.0) - 1.0);
+    }
+
+    solar_panel_voltage = 5.0;
+
+    solar_panel_voltage +=
+            2.5 * electric_power_supply_amp_40 * sin(2 * M_PI * ((electric_power_supply_step_count % 40) * 0.025));
+    solar_panel_voltage +=
+            2.5 * electric_power_supply_amp_20 * sin(2 * M_PI * ((electric_power_supply_step_count % 20) * 0.05));
+    solar_panel_voltage +=
+            2.5 * electric_power_supply_amp_10 * sin(2 * M_PI * ((electric_power_supply_step_count % 10) * 0.1));
+    solar_panel_voltage +=
+            2.5 * electric_power_supply_amp_5 * sin(2 * M_PI * ((electric_power_supply_step_count % 5) * 0.2));
+
+    double power = sin(M_PI * solar_panel_pwm * solar_panel_pwm);
+    solar_panel_current = power * solar_panel_voltage / 20.0;
+
+    double outpower = flow_rate_value * 3.0;
+
+    battery_capacity += (power - outpower) / 1000.0;
+
+    if (battery_capacity < 0.0) {
+        battery_capacity = 0.0;
+    }
+    if (battery_capacity > 1.0) {
+        battery_capacity = 1.0;
+    }
+
+    ++electric_power_supply_step_count;
+    if (electric_power_supply_step_count == 40) {
+        electric_power_supply_step_count = 0;
+    }
+
+    if (payload_step_count == 0) {
+        payload_amp_100 = 0.5 * (rand() / (RAND_MAX - 2.0) - 1.0);
+    }
+
+    if (payload_step_count % 50 == 0) {
+        payload_amp_50 = 0.4 * (rand() / (RAND_MAX - 2.0) - 1.0);
+    }
+
+    if (payload_step_count % 25 == 0) {
+        payload_amp_25 = 0.2 * (rand() / (RAND_MAX - 2.0) - 1.0);
+    }
+
+    payload_flow_rate = payload_target_flow_rate * 20;
+
+    payload_temperature = 305.0 + 20.0 * payload_amp_100 * sin(2.0 * M_PI * ((payload_step_count % 100) * 0.001))
+            + 20.0 * payload_amp_50 * sin(2.0 * M_PI * ((payload_step_count % 50) * 0.02))
+            + 20.0 * payload_amp_25 * sin(2.0 * M_PI * ((payload_step_count % 25) * 0.04)) + payload_flow_rate / 5.0;
+
+    ++payload_step_count;
+    if (payload_step_count == 100) {
+        payload_step_count = 0;
+    }
+}
+
+useconds_t calculate_sleep_time(struct timeval *start_tv, struct timeval *end_tv)
+{
+    if (start_tv->tv_sec == end_tv->tv_sec) {
+        int usec = end_tv->tv_usec - start_tv->tv_usec;
+
+        return 1000000L - usec;
```

Comment:
It would be nice if 100000L was a named symbol

Reply:
Done (there was a mistake, its fixed now)